### ISSUE TITLE
Support for macOS

### ISSUE THREAD

**Issue by Cubitect** [Cubitect/cubiomes-viewer#107]
I created this issue to host a discussion regarding support for macOS.

**I have no method of building for macOS, nor a way to verify pre-built binaries.** For these reasons I will not be supplying official builds for that platform.

There have been offers to contribute by supplying compiled binaries, which users can do so here if they wish. However, my general recommendation would be to compile the application yourself.

The necessary build utilities and Qt dependency can be gotten with XCode and [Homebrew](https://brew.sh):
```
$ xcode-select --install
$ brew install qt@5
$ brew link qt@5
```
(Thanks @rlvandaveer for these instructions)

**Comment by rlvandaveer** [comment]
I am curious how releases are created now. I don't see an Actions. It would not be a heavy lift to add a MacOS stage to an Action. Linux, Windows, and Mac could all be built in a [matrix](https://docs.github.com/en/actions/using-jobs/using-a-build-matrix-for-your-jobs#overview)

**Comment by Cubitect** [comment]
I'm pretty sure GitHub Actions do not offer static Qt builds, since that requires compiling Qt from source and takes around 10GB of space.

Currently, I have a couple of basic VMs specifically setup to build the Linux and Windows releases. So far I hadn't really considered Actions, as I didn't think that it was worth the trouble to automate something that I only need about once a month.

**Comment by emmekappa** [comment]
@Cubitect have you seen this community action? https://github.com/marketplace/actions/install-qt

**Comment by Cubitect** [comment]
Yes, and it doesn't support static builds: [github.com/jurplel/install-qt-action/issues/53](https://github.com/jurplel/install-qt-action/issues/53)

**Comment by TomerGamerTV** [comment]
For anyone asking how to do it just install it as he said above, visit [this](https://github.com/Cubitect/cubiomes-viewer/blob/trunk/buildguide.md) link and scroll down it says how to do it

**Comment by Dax911** [comment]
Idk if anyone needs to know this but for future reference, all versions from August 2022 to now have worked great on M2 apple silicon.  Also, thanks for the updates love this tool.  I want to complain that the default font is American Typewriter. Gross.  Would love to see a tutorial somewhere.

**Comment by ltecheroffical** [comment]
That renders me running a pre-compiled program useless

**Comment by GrantedSuper** [comment]
I am looking to download and use cubiomes viewer for some seed-finding on M1 MacBook Air with no idea how. I am very unfamiliar with Github and am looking for where to download the application for Mac. @Dax911 said they are using it on M2. I would love to know how you did that. If someone could link me to instructions on how to use this for Mac that would be much appreciated.

**Comment by TomerGamerTV** [comment]
> I am looking to download and use cubiomes viewer for some seed-finding on M1 MacBook Air with no idea how. I am very unfamiliar with Github and am looking for where to download the application for Mac. @Dax911 said they are using it on M2. I would love to know how you did that. If someone could link me to instructions on how to use this for Mac that would be much appreciated.

Do you know how to scroll up? Read my comment

**Comment by Dax911** [comment]
> I am looking to download and use cubiomes viewer for some seed-finding on M1 MacBook Air with no idea how. I am very unfamiliar with Github and am looking for where to download the application for Mac. @Dax911 said they are using it on M2. I would love to know how you did that. If someone could link me to instructions on how to use this for Mac that would be much appreciated.

I compiled it from source myself ish I will need to go thru the process again, but it is still working with the newer updates. I have appts today and a new sprint to setup for. But I will make this a priority this week. Sorry everyone.

**Comment by devjta** [comment]
> I am looking to download and use cubiomes viewer for some seed-finding on M1 MacBook Air with no idea how. I am very unfamiliar with Github and am looking for where to download the application for Mac. @Dax911 said they are using it on M2. I would love to know how you did that. If someone could link me to instructions on how to use this for Mac that would be much appreciated.

Get homebrew, install qt5, install xcode (like the guide tells for mac os) and you can build it without ANY OTHER change..

![image](https://user-images.githubusercontent.com/5166270/216980463-d6225897-5789-4c53-98e6-649ec33208f8.png)

**Comment by mark-d-holmberg** [comment]
Can confirm. This worked for me.

**Comment by Dax911** [comment]
Hey I am back and with this latest version I am getting a very interesting compile issue.
```zsh
In file included from noise.c:2:
In file included from ./noise.h:4:
./rng.h:6:10: fatal error: 'stdlib.h' file not found
#include <stdlib.h>
         ^~~~~~~~~~
1 error generated.
make[1]: *** [noise.o] Error 1
make: *** [cubiomes-viewer.app/Contents/MacOS/cubiomes-viewer] Error 2
```
Fixed it on M2

Run 
```zsh
xcrun --show-sdk-path
```
to get the path and make sure ur sdk is up to date and everything for qt is in the correct path etc.

using the output i replaced ln 4 in the makefile as so 

```zsh
override CFLAGS += -Wall -Wextra -fwrapv -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk
```
then cleared everything and started build from scratch.

**Comment by Dax911** [comment]
If there is significant interest I will happily take over supplying compiled Mac binaries for the community.

**Comment by h4nkyn** [comment]
Well I'm interested!

**Comment by samon1210** [comment]
@Dax911 I would like to know more details from the makefile area.

**Comment by TomerGamerTV** [comment]
> Hey I am back and with this latest version I am getting a very interesting compile issue.
> 
> ```shell
> In file included from noise.c:2:
> In file included from ./noise.h:4:
> ./rng.h:6:10: fatal error: 'stdlib.h' file not found
> #include <stdlib.h>
>          ^~~~~~~~~~
> 1 error generated.
> make[1]: *** [noise.o] Error 1
> make: *** [cubiomes-viewer.app/Contents/MacOS/cubiomes-viewer] Error 2
> ```
> 
> Fixed it on M2
> 
> Run
> 
> ```shell
> xcrun --show-sdk-path
> ```
> 
> to get the path and make sure ur sdk is up to date and everything for qt is in the correct path etc.
> 
> using the output i replaced ln 4 in the makefile as so
> 
> ```shell
> override CFLAGS += -Wall -Wextra -fwrapv -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk
> ```
> 
> then cleared everything and started build from scratch.

i tried that but it didn't work

`./rng.h:6:10: fatal error: 'stdlib.h' file not found
#include <stdlib.h>
         ^~~~~~~~~~
1 error generated.
make[1]: *** [noise.o] Error 1
make: *** [cubiomes-viewer.app/Contents/MacOS/cubiomes-viewer] Error 2
`

**Comment by samon1210** [comment]
> > Hey I am back and with this latest version I am getting a very interesting compile issue.
> > ```
> > In file included from noise.c:2:
> > In file included from ./noise.h:4:
> > ./rng.h:6:10: fatal error: 'stdlib.h' file not found
> > #include <stdlib.h>
> >          ^~~~~~~~~~
> > 1 error generated.
> > make[1]: *** [noise.o] Error 1
> > make: *** [cubiomes-viewer.app/Contents/MacOS/cubiomes-viewer] Error 2
> > ```
> > 
> > 
> > Fixed it on M2
> > Run
> > ```
> > xcrun --show-sdk-path
> > ```
> > 
> > 
> > to get the path and make sure ur sdk is up to date and everything for qt is in the correct path etc.
> > using the output i replaced ln 4 in the makefile as so
> > ```
> > override CFLAGS += -Wall -Wextra -fwrapv -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk
> > ```
> > 
> > 
> > then cleared everything and started build from scratch.
> 
> i tried that but it didn't work
> 
> `./rng.h:6:10: fatal error: 'stdlib.h' file not found #include <stdlib.h> ^~~~~~~~~~ 1 error generated. make[1]: *** [noise.o] Error 1 make: *** [cubiomes-viewer.app/Contents/MacOS/cubiomes-viewer] Error 2 `

Yes, I got the same result as you.

**Comment by Dax911** [comment]
Let me walk thru it again, sorry I really should increase the notifications for this thread.

**Comment by samon1210** [comment]
<img width="1728" alt="スクリーンショット 2023-11-06 17 53 24" src="https://github.com/Cubitect/cubiomes-viewer/assets/106953247/d5074b2f-b845-43bc-aed8-3318bb8c1082">
Using wineskin was very easy!

**Comment by samon1210** [comment]
> <img alt="スクリーンショット 2023-11-06 17 53 24" width="1728" src="https://user-images.githubusercontent.com/106953247/280647602-d5074b2f-b845-43bc-aed8-3318bb8c1082.png"> Using wineskin was very easy!

But threads can't be more than 4

**Comment by TomerGamerTV** [comment]
> <img alt="スクリーンショット 2023-11-06 17 53 24" width="1728" src="https://user-images.githubusercontent.com/106953247/280647602-d5074b2f-b845-43bc-aed8-3318bb8c1082.png"> Using wineskin was very easy!

using emulators is not a good idea because it decreases performance drastically, it's possible to make it run natively on Mac I just can't figure out how to compile it.

**Comment by BestBoyBerlin** [comment]
I got the same errors but fixed them by running:
`export	CPATH=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include`
like described here: [https://developer.apple.com/forums/thread/123997](url)
It worked even without editing the Makefile.

**Comment by Dax911** [comment]
> I got the same errors but fixed them by running: `export CPATH=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include` like described here: [https://developer.apple.com/forums/thread/123997](url) It worked even without editing the Makefile.

This seems correct to me I encounter a new compile issue as well w the new version of mac os and this seems to have resolved it. Thank you.

**Comment by TomerGamerTV** [comment]
> I got the same errors but fixed them by running: `export CPATH=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include` like described here: [https://developer.apple.com/forums/thread/123997](url) It worked even without editing the Makefile.

i tried that but after running make it just stops with this code: (this is what it spills at the end this is not the full output )

```
k IOKit -framework OpenGL -framework AGL
ld: warning: object file (/Users/tomer/Downloads/cubiomes-viewer/cubiomes/libcubiomes.a[2](noise.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
ld: warning: object file (/Users/tomer/Downloads/cubiomes-viewer/cubiomes/libcubiomes.a[3](layers.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
ld: warning: object file (/Users/tomer/Downloads/cubiomes-viewer/cubiomes/libcubiomes.a[4](biometree.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
ld: warning: object file (/Users/tomer/Downloads/cubiomes-viewer/cubiomes/libcubiomes.a[5](biomenoise.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
ld: warning: object file (/Users/tomer/Downloads/cubiomes-viewer/cubiomes/libcubiomes.a[6](generator.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
ld: warning: object file (/Users/tomer/Downloads/cubiomes-viewer/cubiomes/libcubiomes.a[7](finders.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
ld: warning: object file (/Users/tomer/Downloads/cubiomes-viewer/cubiomes/libcubiomes.a[8](util.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
ld: warning: object file (/Users/tomer/Downloads/cubiomes-viewer/cubiomes/libcubiomes.a[9](quadbase.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
```

**Comment by BestBoyBerlin** [comment]
> > I got the same errors but fixed them by running: `export CPATH=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include` like described here: [https://developer.apple.com/forums/thread/123997](url) It worked even without editing the Makefile.
> 
> i tried that but after running make it just stops with this code: (this is what it spills at the end this is not the full output )
> 
> ```
> k IOKit -framework OpenGL -framework AGL
> ld: warning: object file (/Users/tomer/Downloads/cubiomes-viewer/cubiomes/libcubiomes.a[2](noise.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
> ld: warning: object file (/Users/tomer/Downloads/cubiomes-viewer/cubiomes/libcubiomes.a[3](layers.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
> ld: warning: object file (/Users/tomer/Downloads/cubiomes-viewer/cubiomes/libcubiomes.a[4](biometree.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
> ld: warning: object file (/Users/tomer/Downloads/cubiomes-viewer/cubiomes/libcubiomes.a[5](biomenoise.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
> ld: warning: object file (/Users/tomer/Downloads/cubiomes-viewer/cubiomes/libcubiomes.a[6](generator.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
> ld: warning: object file (/Users/tomer/Downloads/cubiomes-viewer/cubiomes/libcubiomes.a[7](finders.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
> ld: warning: object file (/Users/tomer/Downloads/cubiomes-viewer/cubiomes/libcubiomes.a[8](util.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
> ld: warning: object file (/Users/tomer/Downloads/cubiomes-viewer/cubiomes/libcubiomes.a[9](quadbase.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
> ```
You're right, but these are just warnings, so it's fine. You should find a cubiomes-viewer.app file in your build folder after running make. Just double click on it and cubiomes-viewer should start.

**Comment by TomerGamerTV** [comment]
> > > I got the same errors but fixed them by running: `export CPATH=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include` like described here: [https://developer.apple.com/forums/thread/123997](url) It worked even without editing the Makefile.
> > 
> > 
> > i tried that but after running make it just stops with this code: (this is what it spills at the end this is not the full output )
> > ```
> > k IOKit -framework OpenGL -framework AGL
> > ld: warning: object file (/Users/tomer/Downloads/cubiomes-viewer/cubiomes/libcubiomes.a[2](noise.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
> > ld: warning: object file (/Users/tomer/Downloads/cubiomes-viewer/cubiomes/libcubiomes.a[3](layers.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
> > ld: warning: object file (/Users/tomer/Downloads/cubiomes-viewer/cubiomes/libcubiomes.a[4](biometree.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
> > ld: warning: object file (/Users/tomer/Downloads/cubiomes-viewer/cubiomes/libcubiomes.a[5](biomenoise.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
> > ld: warning: object file (/Users/tomer/Downloads/cubiomes-viewer/cubiomes/libcubiomes.a[6](generator.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
> > ld: warning: object file (/Users/tomer/Downloads/cubiomes-viewer/cubiomes/libcubiomes.a[7](finders.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
> > ld: warning: object file (/Users/tomer/Downloads/cubiomes-viewer/cubiomes/libcubiomes.a[8](util.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
> > ld: warning: object file (/Users/tomer/Downloads/cubiomes-viewer/cubiomes/libcubiomes.a[9](quadbase.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
> > ```
> 
> You're right, but these are just warnings, so it's fine. You should find a cubiomes-viewer.app file in your build folder after running make. Just double click on it and cubiomes-viewer should start.

Problem is I can’t find it, it just doesn’t exist.

**Comment by markovmagnifico** [comment]
> I got the same errors but fixed them by running: `export CPATH=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include` like described here: [https://developer.apple.com/forums/thread/123997](url) It worked even without editing the Makefile.

Something like this worked for me but I had to run `export CPATH=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include` instead.

I also saw these same errors:
```
ld: warning: object file (/Users/alexcoda/workspace/cubiomes-viewer/cubiomes/libcubiomes.a(noise.o)) was built for newer macOS version (13.0) than being linked (11.0)
ld: warning: object file (/Users/alexcoda/workspace/cubiomes-viewer/cubiomes/libcubiomes.a(layers.o)) was built for newer macOS version (13.0) than being linked (11.0)
ld: warning: object file (/Users/alexcoda/workspace/cubiomes-viewer/cubiomes/libcubiomes.a(biometree.o)) was built for newer macOS version (13.0) than being linked (11.0)
ld: warning: object file (/Users/alexcoda/workspace/cubiomes-viewer/cubiomes/libcubiomes.a(biomenoise.o)) was built for newer macOS version (13.0) than being linked (11.0)
ld: warning: object file (/Users/alexcoda/workspace/cubiomes-viewer/cubiomes/libcubiomes.a(generator.o)) was built for newer macOS version (13.0) than being linked (11.0)
ld: warning: object file (/Users/alexcoda/workspace/cubiomes-viewer/cubiomes/libcubiomes.a(finders.o)) was built for newer macOS version (13.0) than being linked (11.0)
ld: warning: object file (/Users/alexcoda/workspace/cubiomes-viewer/cubiomes/libcubiomes.a(util.o)) was built for newer macOS version (13.0) than being linked (11.0)
ld: warning: object file (/Users/alexcoda/workspace/cubiomes-viewer/cubiomes/libcubiomes.a(quadbase.o)) was built for newer macOS version (13.0) than being linked (11.0)
```

But the cubiomes-viewer.app generated for me and works fine!

For the record I'm on an M1 mac, macOS 13.4.1

**Comment by Cubitect** [comment]
I've set up a github action to deploy a dmg for the newest release. I still don't have a way to check that it actually runs, though.
Does someone want to confirm if it works, or provide detail on potential errors?

**Comment by Dax911** [comment]
Sure thing

**Comment by Dax911** [comment]
Not working, something to do w Qt plugin

```bash

Last login: Sat Feb 10 00:00:24 on ttys001
You have new mail.
/Users/dax/.zshrc:source:113: no such file or directory: /opt/homebrew/opt/powerlevel10k/powerlevel10k.zsh-theme
/Users/dax/.zshrc:source:114: no such file or directory: /opt/homebrew/opt/powerlevel10k/powerlevel10k.zsh-theme
/Volumes/cubiomes-viewer/cubiomes-viewer.app/Contents/MacOS/cubiomes-viewer ; exit;
➜  ~ /Volumes/cubiomes-viewer/cubiomes-viewer.app/Contents/MacOS/cubiomes-viewer ; exit;
qt.qpa.plugin: Could not find the Qt platform plugin "cocoa" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

[1]    4786 abort      /Volumes/cubiomes-viewer/cubiomes-viewer.app/Contents/MacOS/cubiomes-viewer

Saving session...completed.
Deleting expired sessions...       8 completed.

[Process completed]

```

**Comment by nealxm** [comment]
i have tried building cubiomes from source in the past so i have all the qt dependencies but i was never successful in building. i also did not have success running your posted dmg (both v4.0.0 and v4.0.1), here is the full crash log: https://pastebin.com/QviugsdA

EDIT:
since you've added some stuff for the macos build and i have not tried building from source in a while, i just tried to build again and everything worked! let me know if i can do anything to help with the official build

**Comment by devjta** [comment]
I really dont know, what you are all doing..

![image](https://github.com/Cubitect/cubiomes-viewer/assets/5166270/e203f97b-f826-405b-92cd-b728f256c11e)

Build on Apple M1 Max (MBP 14" 2021).

**Comment by MonotoneDevelopment** [comment]
I'm on an Intel and it doesn't work, the crash trace says the program called abort().

**Comment by MonotoneDevelopment** [comment]
> > <img alt="スクリーンショット 2023-11-06 17 53 24" width="1728" src="https://user-images.githubusercontent.com/106953247/280647602-d5074b2f-b845-43bc-aed8-3318bb8c1082.png"> Using wineskin was very easy!
> 
> using emulators is not a good idea because it decreases performance drastically, it's possible to make it run natively on Mac I just can't figure out how to compile it.

As seen before, it runs on Wine. One thing to note, Wine is not an emulator (Which is what Wine stands for), and instead it runs it through a compatibility layer to effectively make it run with no overhead from an emulator, and instead run it as if it was a native app. That means it should be a viable solution/option.

**Comment by LucasRapinoja** [comment]
I know very little about this stuff, but I have tried building from scratch and the dmg. Building from scratch I got everything built but when I try to open the app it says it cant be opened because it's damaged or incomplete. The dmg seemed a bit more promising, but it crashes immediately. If I go into the package contents, open Contents> Mac OS and run the cubiomes viewer executable I get this:

> qt.qpa.plugin: Could not find the Qt platform plugin "cocoa" in ""
> This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
> 
> zsh: abort      /Volumes/cubiomes-viewer/cubiomes-viewer.app/Contents/MacOS/cubiomes-viewer

I would really appreciate if someone could help. Also I'm on a Mac with Apple Silicon and I'm on Sonoma (14.5)

**Comment by devjta** [comment]
Tried it again just now.. 

Install qt and xcode

git clone https://github.com/Cubitect/cubiomes-viewer.git
then going into the directory, then

git submodule update --init cubiomes
mkdir build
cd build
qmake ..
make
open cubiomes-viewer.app

![image](https://github.com/user-attachments/assets/ae589a5a-f9ac-4c1f-b2f6-81aeed64aee9)

**Comment by rmart1n-dev** [comment]
> Tried it again just now..
> 
> Install qt and xcode
> 
> git clone https://github.com/Cubitect/cubiomes-viewer.git then going into the directory, then
> 
> git submodule update --init cubiomes mkdir build cd build qmake .. make open cubiomes-viewer.app
> 
> ![image](https://private-user-images.githubusercontent.com/5166270/351750162-ae589a5a-f9ac-4c1f-b2f6-81aeed64aee9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzExODUwNzgsIm5iZiI6MTczMTE4NDc3OCwicGF0aCI6Ii81MTY2MjcwLzM1MTc1MDE2Mi1hZTU4OWE1YS1mOWFjLTRjMWYtYjJmNi04MWFlZWQ2NGFlZTkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMDlUMjAzOTM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGUwZWUzOTQ4MjNhZjcyMjRjZjk5OTU1NzE1OTc4MzZjYjc2Y2I4MGI1MDgwMGFmMTA0MWZjNDQ5MDljYWNlNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.8xSLc2kNd38oviB8lHOVWcP9-xAjkn9z38HEuGgMlVc)

This isn't working for me. Followed all instructions, getting this error.

`/opt/homebrew/Cellar/qt@5/5.15.15_2/lib/QtCore.framework/Headers/qglobal.h:45:12: fatal error: 'type_traits' file not found
   45 | #  include <type_traits>
      |            ^~~~~~~~~~~~~
1 error generated.
make: *** [aboutdialog.o] Error 1`

**Comment by devjta** [comment]
Yeah, mine would also have that include statement, but its not building for me now anyway.

`../src/search.h:677:28: error: implicit instantiation of undefined template 'std::vector<Condition>'
    std::vector<Condition> condvec;
                           ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/iosfwd:259:28: note: template is declared here
class _LIBCPP_TEMPLATE_VIS vector;
                           ^
In file included from ../src/headless.cpp:1:
In file included from ../src/headless.h:4:
In file included from ../src/searchthread.h:4:
../src/search.h:678:36: error: implicit instantiation of undefined template 'std::vector<std::vector<char>>'
    std::vector<std::vector<char>> references;
                                   ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/iosfwd:259:28: note: template is declared here
class _LIBCPP_TEMPLATE_VIS vector;
                           ^
../src/headless.cpp:61:5: error: no matching member function for call to 'connect'
    connect(&timer, &QTimer::timeout, this, QOverload<>::of(&Headless::progressTimeout));`

But I am still at Mac OS 12.5... Let me upgrade to at least 14.

**Comment by devjta** [comment]
So, even upgraded to latest Mac OS. Needed to run xcode-select --install and brew upgrade, but at least its building fine again.

![image](https://github.com/user-attachments/assets/35afcd16-eb07-4e25-97cc-766bc8fc5efa)
![image](https://github.com/user-attachments/assets/8afa6b70-2247-461d-b8d5-02fa91fb8515)

**Comment by mathuin** [comment]
Builds great on my M4 following the instructions above, though it appears to only show Java edition and not Bedrock.  Going to look further into the docs and code to see why.

**Comment by Dax911** [comment]
> Builds great on my M4 following the instructions above, though it appears to only show Java edition and not Bedrock. Going to look further into the docs and code to see why.

I thought this app supports Java Edition only/primarily.

**Comment by mathuin** [comment]
https://mcseedmap.net (which supports both Java and Bedrock) is "powered by" cubiomes among others, and recommends this tool for advanced viewing, so I thought I would give it a try.

**Comment by Dax911** [comment]
> https://mcseedmap.net (which supports both Java and Bedrock) is "powered by" cubiomes among others, and recommends this tool for advanced viewing, so I thought I would give it a try.

Yea this application allows you to set search parameters and actually start brute forcing thru all seeds that meet your criteria.

**Comment by Aero-Blue** [comment]
Just an update for those using mac. I am on Sequoia on an Macbook Pro (M1 Pro chip). The builds from 4+ onwards seem to be broken for mac. I could open previous ones but they would crash immediately. You have to build from source in order for it to work, at least in my experience the steps above do not work (initially provided at the top).

Here is a simple step by step guide to show how to build from source on mac, as the releases (.dmg)'s don't seem to work.

You do still need to install xcode, and qt@5 (so follow the initial steps at the top).
`git clone --recursive https://github.com/Cubitect/cubiomes-viewer.git`
`cd cubiomes-viewer`
`mkdir build && cd build`
`qmake ..` * There will be some warnings most likely, but no errors.
`make`
`mv cubiomes-viewer.app /Applications`

You now have cubiomes-viewer listed in your apps. You will need to open it twice, and allow the unknown app through some location that's similar to "Privacy & Security". After closing the dialog, you confirm the app in settings, and then open it again.

**Comment by sneznaj** [comment]
you can get an ai bot that uses the command line for you. thats how i did it.

**Comment by sneznaj** [comment]
i prompted together  a .dmg for macos with bedrock 1.6 and 1.7 support. https://github.com/sneznaj/cubiomes-viewer-macos/releases/tag/macos
